cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java

```diff
@@ -498,16 +512,22 @@ public void getArtifactPlugin(HttpRequest request, HttpResponder responder,
                                 @PathParam("artifact-version") String artifactVersion,
                                 @PathParam("plugin-type") String pluginType,
                                 @PathParam("plugin-name") String pluginName,
-                                @QueryParam("scope") @DefaultValue("user") String scope)
-    throws NamespaceNotFoundException, BadRequestException, ArtifactNotFoundException {
+                                @QueryParam("scope") @DefaultValue("user") String scope,
+                                @QueryParam("versionRange") String versionRange,
+                                @QueryParam("limit") @DefaultValue("-1") String limit,
+                                @QueryParam("order") String order)
+    throws NamespaceNotFoundException, BadRequestException, ArtifactNotFoundException, InvalidArtifactRangeException {
 
     NamespaceId namespace = Ids.namespace(namespaceId);
     NamespaceId artifactNamespace = validateAndGetScopedNamespace(namespace, scope);
     Id.Artifact artifactId = validateAndGetArtifactId(artifactNamespace, artifactName, artifactVersion);
+    ArtifactRange range = versionRange == null ? null : ArtifactRange.parse(namespace, artifactName, versionRange);
+    int limitNumber = Integer.valueOf(limit);
+    SortInfo.SortOrder sortOrder = order == null ? null : SortInfo.SortOrder.valueOf(order);
 
     try {
       SortedMap<ArtifactDescriptor, PluginClass> plugins =
-        artifactRepository.getPlugins(namespace, artifactId, pluginType, pluginName);
+        artifactRepository.getPlugins(namespace, artifactId, pluginType, pluginName, range, limitNumber, sortOrder);
```

Comment:
I think we are already doing this. In this `getPlugins()` method, namespace is a scoped namespace based on the query-param scope, plugin-name is the plugin name, and range can be either range or an exact version.